src/main/java/com/arpnetworking/tsdcore/model/CompositeUnit.java

```diff
@@ -0,0 +1,338 @@
+         */
+        ZEPTO(10 ^ -21),
+        /**
+         * 10^-18.
+         */
+        ATTO(10 ^ -18),
+        /**
+         * 10^-15.
+         */
+        FEMTO(10 ^ -15),
+        /**
+         * 10^-12.
+         */
+        PICO(10 ^ -12),
+        /**
+         * 10^-9.
+         */
+        NANO(10 ^ -9),
+        /**
+         * 10^-6.
+         */
+        MICRO(10 ^ -6),
+        /**
+         * 10^-3.
+         */
+        MILLI(10 ^ -3),
+        /**
+         * 10^-2.
+         */
+        CENTI(10 ^ -2),
+        /**
+         * 10^-1.
+         */
+        DECI(10 ^ -1),
+        /**
+         * 1.
+         */
+        ONE(1),
+        /**
+         * 10.
+         */
+        DECA(10),
+        /**
+         * 10^2.
+         */
+        HECTO(10 ^ 2),
+        /**
+         * 10^3.
+         */
+        KILO(10 ^ 3),
+        /**
+         * 10^6.
+         */
+        MEGA(10 ^ 6),
+        /**
+         * 10^9.
+         */
+        GIGA(10 ^ 9),
+        /**
+         * 10^12.
+         */
+        TERA(10 ^ 12),
+        /**
+         * 10^15.
+         */
+        PETA(10 ^ 15),
+        /**
+         * 10^18.
+         */
+        EXA(10 ^ 18),
+        /**
+         * 10^21.
+         */
+        ZETTA(10 ^ 21),
+        /**
+         * 10^24.
+         */
+        YOTTA(10 ^ 24),
+        /**
+         * 10^27.
+         */
+        KIBI(10 ^ 27),
+        /**
+         * 10^30.
+         */
+        MEBI(10 ^ 30),
+        /**
+         * 10^33.
+         */
+        GIBI(10 ^ 33),
+        /**
+         * 10^36.
+         */
+        TEBI(10 ^ 36),
+        /**
+         * 10^39.
+         */
+        PEBI(10 ^ 39),
+        /**
+         * 10^42.
+         */
+        EXBI(10 ^ 42),
+        /**
+         * 10^45.
+         */
+        ZEBI(10 ^ 45),
+        /**
+         * 10^48.
+         */
+        YOBI(10 ^ 48);
+
+        /**
+         * Constructor for Scale class.
+         *
+         * @param value numerical value of Scale for multiplication
+         */
+        Scale(final long value) {
+            _value = value;
+        }
+
+        private long _value;
+    }
+
+    enum Category {
+        TIME,
+        DATA_SIZE,
+        TEMPERATURE,
+        ROTATION
+    }
+
+    @NotNull
+    private Type _type;
+    private Scale _scale;
+
+    /**
+     * Get the existing unit that maps to the covmmpound unit.
+     *
+     * @param compositeUnit
+     * ompoundUnit newer unit format.
+     * @return existing unit which maps to the newer format
+     */
+    public static Unit getLegacyUnit(final Optional<CompositeUnit> compositeUnit) {
+        return compositeUnit.isPresent()
+                ? LEGACY_UNIT_MAP.getOrDefault(compositeUnit.get(), null)
+                : null;
+    }
+
+    private static final ImmutableMap<CompositeUnit, Unit> LEGACY_UNIT_MAP = new ImmutableMap.Builder<CompositeUnit, Unit>()
+            .put(new CompositeUnit(Scale.NANO, Type.SECOND), Unit.NANOSECOND)
```

Comment:
As I mentioned in the previous review, Brandon did this by simply concatenation the scale and unit. While not pretty as far as I know it worked just fine. I'm happy to wrap that in a function to allow us to change the logic in one place. However this mapping should be in the parser classes.